### PR TITLE
CP-52114: Add pool.license_server for pool level licensing

### PR DESCRIPTION
**Pull request by changlei-li**
Add new XAPI pool level data for licensing

### CODE REVIEW

File: ocaml/idl/datamodel_pool.ml [hunk 1]
```diff
@@ -2153,6 +2153,10 @@ let t =
             ~default_value:(Some (VMap [])) "recommendations"
             "The recommended pool properties for clients to respect for \
              optimal performance. e.g. max-vm-group=5"
+        ; field ~writer_roles:_R_POOL_OP ~qualifier:RW ~lifecycle:[]
+            ~ty:(Map (String, String))
+            ~default_value:(Some (VMap [])) "license_server"
+            "licensing data shared with the whole pool"
```
Comment (gangj) [inline ocaml/idl/datamodel_pool.ml:2159]: Licensing
with -> within?

**Review by gangj — APPROVED** [review]

File: ocaml/xapi-cli-server/records.ml [hunk 2]
```diff
@@ -1494,6 +1494,18 @@ let pool_record rpc session_id pool =
...
+          ~add_to_map:(fun key value ->
+            Client.Pool.add_to_license_server ~rpc ~session_id ~self:pool ~key
+              ~value
+          )
+          ~remove_from_map:(fun key ->
+            Client.Pool.remove_from_license_server ~rpc ~session_id ~self:pool
+              ~key
+          )
+          ~get_map:(fun () -> (x ()).API.pool_license_server)
+          ()
```
Comment (minglumlu) [inline ocaml/xapi-cli-server/records.ml:1508]: I think a `set_in_map` would be useful as well.

File: ocaml/idl/datamodel_pool.ml (hunk 1 above)
Comment (changlei-li) [reply ocaml/idl/datamodel_pool.ml:2159]: accept

File: ocaml/xapi-cli-server/records.ml (hunk 2 above)
Comment (changlei-li) [reply ocaml/xapi-cli-server/records.ml:1508]: when `set_in_map` is not set, `set_field` will use `remove_from_map` then `add_to_map` to set map pair. So, command like `xxe pool-param-set uuid=38abeb4c-7a61-7ef2-32a9-35e71cfb0fdf license-server:address=192.168.1.2` is already available.

File: ocaml/idl/datamodel_pool.ml (hunk 1 above)
Comment (changlei-li) [reply ocaml/idl/datamodel_pool.ml]: done

**Review by minglumlu — APPROVED** [review]

**Review by robhoes — APPROVED** [review]

**Pull request merged**